jerry-core/ecma/operations/ecma-iterator-object.c

```diff
@@ -242,83 +239,27 @@ ecma_op_get_iterator (ecma_value_t value, /**< value to get iterator from */
     return ecma_raise_type_error (ECMA_ERR_MSG ("Iterator is not an object"));
   }
 
-  if (next_method_p != NULL)
-  {
-    ecma_object_t *obj_p = ecma_get_object_from_value (iterator);
-    ecma_value_t next_method = ecma_op_object_get_by_magic_id (obj_p, LIT_MAGIC_STRING_NEXT);
-
-    if (ECMA_IS_VALUE_ERROR (next_method))
-    {
-      ecma_free_value (iterator);
-      return next_method;
-    }
-
-    if (ecma_is_value_object (next_method) && ecma_op_is_callable (next_method))
-    {
-      *next_method_p = next_method;
-    }
-    else
-    {
-      ecma_free_value (next_method);
-    }
-  }
-
-  /* 6. */
-  return iterator;
-} /* ecma_op_get_iterator */
-
-/**
- * IteratorNext operation
- *
- * See also: ECMA-262 v6, 7.4.2
- *
- * Note:
- *      Returned value must be freed with ecma_free_value.
- *
- * @return iterator result object - if success
- *         raised error - otherwise
- */
-static ecma_value_t
-ecma_op_iterator_next_old (ecma_value_t iterator, /**< iterator value */
-                           ecma_value_t value) /**< the routines's value argument */
-{
-  JERRY_ASSERT (ecma_is_value_object (iterator));
-
-  /* 1 - 2. */
   ecma_object_t *obj_p = ecma_get_object_from_value (iterator);
+  ecma_value_t next_method = ecma_op_object_get_by_magic_id (obj_p, LIT_MAGIC_STRING_NEXT);
 
-  ecma_value_t func_next = ecma_op_object_get_by_magic_id (obj_p, LIT_MAGIC_STRING_NEXT);
-
-  if (ECMA_IS_VALUE_ERROR (func_next))
-  {
-    return func_next;
-  }
-
-  if (!ecma_is_value_object (func_next) || !ecma_op_is_callable (func_next))
+  if (ECMA_IS_VALUE_ERROR (next_method))
   {
-    ecma_free_value (func_next);
-    return ecma_raise_type_error (ECMA_ERR_MSG ("Iterator 'next' is not callable"));
+    ecma_free_value (iterator);
+    return next_method;
   }
 
-  ecma_object_t *next_obj_p = ecma_get_object_from_value (func_next);
-
-  bool has_value = !ecma_is_value_empty (value);
-
-  ecma_value_t result;
-  if (has_value)
+  if (ecma_is_value_object (next_method) && ecma_op_is_callable (next_method))
```

Comment:
AFAIK, we don't need to check if next method is an object, the op_is_callable already does this internally.

Reply:
Good catch. I've also removed it elsewhere.